scripts/contributor_dashboard_debug.py

```diff
@@ -0,0 +1,210 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This script helps develop for the contributor dashboard."""
+
+from __future__ import annotations
+
+import hashlib
+import json
+import os
+
+from core import feconf
+from core.constants import constants
+
+import firebase_admin
+from firebase_admin import auth as firebase_auth
+
+import requests
+from typing import Any, Dict, List
+
+FIREBASE_AUTH_EMULATOR_HOST = 'localhost:%s' % feconf.FIREBASE_EMULATOR_PORT
+os.environ['FIREBASE_AUTH_EMULATOR_HOST'] = FIREBASE_AUTH_EMULATOR_HOST
+FIREBASE_SIGN_IN_URL = (
+    'http://' + FIREBASE_AUTH_EMULATOR_HOST +
+    '/identitytoolkit.googleapis.com/v1/accounts:signInWithPassword')
+
+SUPER_ADMIN_EMAIL = feconf.ADMIN_EMAIL_ADDRESS
+SUPER_ADMIN_USERNAME = 'a'
+SUPER_ADMIN_ROLES = [feconf.ROLE_ID_CURRICULUM_ADMIN,
+        feconf.ROLE_ID_TRANSLATION_ADMIN, feconf.ROLE_ID_QUESTION_ADMIN]
+
+CONTRIBUTOR_EMAIL = 'contributor@example.com'
+CONTRIBUTOR_USERNAME = 'b'
+
+CLASSROOM_NAME = 'math'
+CLASSROOM_URL_FRAGMENT = 'math'
+
+
+class ContributorDashboardDebugRequests():
+    """Client-side requests to help develop for the contributor dashboard."""
+
+    def __init__(self, base_url: str) -> None:
+        self.session = requests.session()
+        self.base_url = base_url
+        self.csrf_token = ''
+
+    def populate_data_for_contributor_dashboard_debug(self) -> None:
+        """Populate sample data to help develop for the contributor
+        dashboard.
+        """
+        firebase_admin.initialize_app(
+            options={'projectId': feconf.OPPIA_PROJECT_ID})
+
+        self.sign_up_new_user(SUPER_ADMIN_EMAIL, SUPER_ADMIN_USERNAME)
+        self.sign_up_new_user(CONTRIBUTOR_EMAIL, CONTRIBUTOR_USERNAME)
+
+        self.begin_session(SUPER_ADMIN_EMAIL)
+        self.csrf_token = self.get_csrf_token()
+        self.assign_admin_roles(SUPER_ADMIN_ROLES, SUPER_ADMIN_USERNAME)
+        self.add_submit_question_rights(CONTRIBUTOR_USERNAME)
+        self.generate_dummy_new_structures_data()
+        self.add_topics_to_classroom(CLASSROOM_NAME, CLASSROOM_URL_FRAGMENT)
+
+    def _make_request(
+        self,
+        method: str,
+        url: str,
+        params: Dict[str, Any] | None = None,
+        headers: Dict[str, Any] | None = None
+        ) -> requests.Response:
+        """Makes a request to the Oppia server."""
+        if params is None:
+            params = {}
+        if headers is None:
+            headers = {}
+
+        response = self.session.request(
+            method, self.base_url + url, headers=headers, params=params)
+
+        return response
+
+    def _sign_in_with_email_and_password(
```

Comment:
Typically order methods by usage. E.g. since this method is only used by `begin_session()`, it should go right below it. If a method is being used in a lot of places, like `_make_request()`, then place it at the very bottom of the file. This makes it easier to read the file going top to bottom: higher-level items go up top, lower-level implementation details are towards the bottom.